lib/src/solver/failure.dart

```diff
@@ -35,14 +37,17 @@ class SolveFailure implements ApplicationException {
     return null;
   }
 
-  SolveFailure(this.incompatibility)
+  SolveFailure(this.incompatibility, {this.suggestions})
       : assert(incompatibility.terms.isEmpty ||
             incompatibility.terms.single.package.isRoot);
 
   /// Describes how [incompatibility] was derived, and thus why version solving
   /// failed.
   @override
-  String toString() => _Writer(incompatibility).write();
+  String toString() => [
+        _Writer(incompatibility).write(),
+        if (suggestions != null) suggestions
+      ].join();
```

Comment:
Are you sure you don't want:
```suggestion
      ].join('\n');
```